WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java

```diff
@@ -155,22 +185,38 @@ public int getItemViewType(int position) {
     @Override
     public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
         Context context = parent.getContext();
-        if (viewType == VIEW_TYPE_SPACER) {
-            View spacerView = new View(context);
-            int toolbarHeight = context.getResources().getDimensionPixelSize(R.dimen.toolbar_height);
-            int dividerHeight = context.getResources().getDimensionPixelSize(R.dimen.reader_card_gutters);
-            int spacerHeight = toolbarHeight - dividerHeight;
-            spacerView.setLayoutParams(new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, spacerHeight));
-            return new SpacerViewHolder(spacerView);
-        } else {
-            View postView = LayoutInflater.from(context).inflate(R.layout.reader_cardview_post, parent, false);
-            return new ReaderPostViewHolder(postView);
+        switch (viewType) {
+            case VIEW_TYPE_SPACER:
+                View spacerView = new View(context);
+                int toolbarHeight = context.getResources().getDimensionPixelSize(R.dimen.toolbar_height);
+                int dividerHeight = context.getResources().getDimensionPixelSize(R.dimen.reader_card_gutters);
+                int spacerHeight = toolbarHeight - dividerHeight;
+                spacerView.setLayoutParams(new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, spacerHeight));
+                return new SpacerViewHolder(spacerView);
+
+            case VIEW_TYPE_BLOG_INFO:
+                return new BlogInfoViewHolder(new ReaderBlogInfoView(context));
+
+            case VIEW_TYPE_TAG_INFO:
+                return new TagInfoViewHolder(new ReaderTagInfoView(context));
+
+            default:
+                View postView = LayoutInflater.from(context).inflate(R.layout.reader_cardview_post, parent, false);
+                return new ReaderPostViewHolder(postView);
         }
     }
 
     @Override
     public void onBindViewHolder(final RecyclerView.ViewHolder holder, final int position) {
-        if (holder instanceof SpacerViewHolder) {
+        if (!(holder instanceof ReaderPostViewHolder)) {
```

Comment:
This reads weird to me but it seems like the best way to handle it since the majority of views will be `ReaderPostViewHolder`.